libnmstate/validator.py

```diff
@@ -91,8 +95,11 @@ def validate_interface_capabilities(ifaces_state, capabilities):
             )
 
 
-def validate_interfaces_state(desired_state, current_state):
-    validate_link_aggregation_state(desired_state, current_state)
+def validate_interfaces_state(
+    original_desired_state, desired_state, current_state
+):
```

Comment:
The reason I pass the `original_desired_state` is because the `desired_state` is changed by:

```python
    desired_state.sanitize_ethernet(current_state)
    desired_state.sanitize_dynamic_ip()
    desired_state.merge_routes(current_state)
    desired_state.merge_dns(current_state)
    desired_state.merge_route_rules(current_state)
    desired_state.remove_unknown_interfaces()
    desired_state.complement_master_interfaces_removal(current_state)
    metadata.generate_ifaces_metadata(desired_state, current_state)
```